editor/src/messages/tool/common_functionality/shape_editor.rs

```diff
@@ -50,6 +50,51 @@ pub struct ManipulatorPointInfo<'a> {
 
 pub type OpposingHandleLengths = HashMap<Vec<LayerId>, HashMap<ManipulatorGroupId, Option<f64>>>;
 
+const OUT_HANDLE: u8 = 0b001;
+const IN_HANDLE: u8 = 0b010;
+const BOTH_HANDLES: u8 = 0b011;
+const ANCHOR: u8 = 0b100;
+const ANCHOR_OUT: u8 = 0b101;
+const ANCHOR_IN: u8 = 0b110;
+const ALL: u8 = 0b111;
+
+#[derive(Clone, Eq, PartialEq)]
+pub struct ManipulatorGroupData {
+	pub selection_type: u8,
+	pub group_id: ManipulatorGroupId,
+	pub layer_id: Vec<LayerId>,
+}
```

Comment:
We already store the selected points in `SelectedLayerState`.

Reply:
All selected point state is now taken as needed from the `SelectedLayerState` and processed as needed.